### PR TITLE
Fix Brutus example on nightly.

### PR DESCRIPTION
**Pull request by jumerckx**
Does this make CI happy?

### CODE REVIEW

File: examples/brutus.jl [hunk 1]
```diff
@@ -143,7 +143,11 @@ function code_mlir(f, types)
         for sidx in bb.stmts
             stmt = ir.stmts[sidx]
             inst = stmt[:inst]
-            line = ir.linetable[stmt[:line]+1]
+            line = @static if VERSION <= v"1.11"
+                ir.linetable[stmt[:line]+1]
+            else
+                (; ((:file, :line) .=> Base.IRShow.debuginfo_firstline(ir.debuginfo))...)
```
Comment (Pangoraw) [inline examples/brutus.jl:149]: IIUC, something like this should be more precise to each statement (though we don't really use locations anyway):
```suggestion
                last(Base.IRShow.buildLineInfoNode(ir.debuginfo, :var"n/a", sidx))
```
Feel free to choose either.

File: examples/brutus.jl [hunk 2]
```diff
@@ -186,7 +190,11 @@ function code_mlir(f, types)
                 cond_br = cf.cond_br(cond, true_args, false_args; trueDest=other_dest, falseDest=dest, location)
                 push!(current_block, cond_br)
             elseif inst isa ReturnNode
-                line = ir.linetable[stmt[:line]+1]
+                line = @static if VERSION < v"1.12"
+                    ir.linetable[stmt[:line]+1]
+                else
+                    (; ((:file, :line) .=> Base.IRShow.debuginfo_firstline(ir.debuginfo))...)
+                end
```
Comment (Pangoraw) [inline examples/brutus.jl:197]: We can just remove this and use the `line` defined above.

```suggestion
```

**Review by Pangoraw — COMMENTED** [review]
Thanks!

**Comment by mofeing** [comment]
Thanks for fixing the `v17` module inclusion. I forgot about that!

**Review by mofeing — APPROVED** [review]

**Comment by jumerckx** [comment]
Nightly still segfaults when trying to execute the code. I can investigate later today

File: examples/brutus.jl (hunk 1 above)
Comment (jumerckx) [reply examples/brutus.jl]: buildLineInfoNode sometimes returns an empty list. I pushed a change that uses `debuginfo_firstline` as a fallback in that case, but maybe there's a cleaner solution...

**Comment by jumerckx** [comment]
CI is green. I updated `Pass.jl` with an additional version check. I believe this is good to go?

**Review by Pangoraw — APPROVED** [review]
Great! Sorry I had forgotten that some instructions don't have line info.

**Pull request merged**